### PR TITLE
Add depinfo

### PR DESCRIPTION
**Pull request by cdiener**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/depinfo/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+    - pipdeptree >=0.13.2
+
+test:
+  imports:
+    - depinfo
+
+about:
+  home: https://github.com/Midnighter/dependency-info
+  license: Apache 2.0
```
Comment (chrisburr) [inline recipes/depinfo/meta.yaml]: For consistency with other recipes (SPDX naming uses a dash):
```suggestion
  license: Apache-2.0
```

File: recipes/depinfo/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    A utility Python package intended for other library packages.
+    Provides a function that when called with your package name,
+    will print platform and dependency information.
+  doc_url: https://github.com/Midnighter/dependency-info
+  dev_url: https://github.com/Midnighter/dependency-info
+
+extra:
+  recipe-maintainers:
+    - cdiener
+    - Midnighter
```
Comment (chrisburr) [inline recipes/depinfo/meta.yaml]: @Midnighter Are you happy to be listed as a maintainer of this recipe?

File: recipes/depinfo/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python
+    - pip
+    - pipdeptree >=0.13.2
```
Comment (chrisburr) [inline recipes/depinfo/meta.yaml]: Is this needed in `host`? Normally `run` is enough:
```suggestion
```

File: recipes/depinfo/meta.yaml (hunk 2 above)
Comment (cdiener) [reply recipes/depinfo/meta.yaml]: Pinged him on gitter and he agreed, may take a little until he replies here.

Comment (Midnighter) [reply recipes/depinfo/meta.yaml]: Yes, I'm on board.

**Pull request merged**